### PR TITLE
Add lowering of `aten.log_softmax` op.

### PR DESCRIPTION
**Pull request by pashu123**
The `aten.log_softmax` is decomposed into `aten.softmax` and
`aten.log` op.

### CODE REVIEW

File: lib/Dialect/Torch/Transforms/RefineTypes.cpp [hunk 1]
```diff
@@ -1271,6 +1276,18 @@ ChangeResult TypeAnalyzer::visitAtenSoftmaxIntOp(
   return getLatticeElement(op.getResult()).join(knowledge);
 }
 
+ChangeResult TypeAnalyzer::visitAtenLogSoftmaxIntOp(
```
Comment (cathyzhyi) [inline lib/Dialect/Torch/Transforms/RefineTypes.cpp:1279]: Shall we reuse `TypeAnalyzer::visitAtenSoftmaxIntOp`?  Here is a similar example using template to refactor the code `https://github.com/llvm/torch-mlir/blob/127c7d8e27ede211af77392b5277ba06fb515fb7/lib/Dialect/Torch/Transforms/RefineTypes.cpp#L326`

**Review by cathyzhyi — CHANGES_REQUESTED** [review]

File: e2e_testing/torchscript/basic.py [hunk 2]
```diff
@@ -512,3 +512,21 @@ def forward(self, x, y):
 @register_test_case(module_factory=lambda: TensorToInt())
 def TensorToInt_basic(module, tu: TestUtils):
     module.forward(torch.randint(10,[]), tu.rand())
+    
+class LogSoftmaxIntModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        torch.manual_seed(0)
```
Comment (ramiro050) [inline e2e_testing/torchscript/basic.py:519]: I noticed that this test and some other tests in this file set the random generator seed to zero. Is there a reason we want these tests to always run on the same input?

Comment (cathyzhyi) [reply e2e_testing/torchscript/basic.py:519]: This is to make sure the golden test and the real test would have the same parameters.

Comment (cathyzhyi) [reply e2e_testing/torchscript/basic.py:519]: actually... softmax doesn't have parameters so this might not be needed.

Comment (pashu123) [reply e2e_testing/torchscript/basic.py]: I see, I have removed the random seed param.

File: lib/Dialect/Torch/Transforms/RefineTypes.cpp (hunk 1 above)
Comment (pashu123) [reply lib/Dialect/Torch/Transforms/RefineTypes.cpp]: Done.

**Review by cathyzhyi — APPROVED** [review]
LGTM!
